source/whats-new.txt

```diff
@@ -57,6 +57,77 @@ Learn what's new in:
 * :ref:`Version 3.7 <version-3.7>`
 * :ref:`Version 3.6 <version-3.6>`
 
+.. _version-6.6:
+
+What's New in 6.6
+-----------------
+
+The {+driver-short+} v6.6 release includes the following features:
+
+- Adds the ``addStage()`` method to the fluid aggregation API. You can use this method to
+  add aggregation pipeline stages individually, as shown in the following
+  example:
+
+  .. code-block:: javascript
+
+     const documents = await users.aggregate().addStage({ $project: { name: true } }).toArray();
+
+- Adds the ``cause`` and ``dependencyName`` fields to the ``MongoMissingDependencyError``
+  class. You can use these fields to programmatically determine if a package is missing
+  or why a package didn't load.
+
+- Adds the ``minRoundTripTime`` property to the ``ServerDescription`` class. This
+  property contains the minimum round-trip time over the last 10 heartbeats.
+
+- Adds the ``toJSON()`` method to the ``TopologyDescription`` class. You can use this
+  method to stringify ``TopologyDescription`` objects to JSON.
+
+- Adds cursor support for the ``Collection.indexExists()``, ``Collection.indexes()``, and
+  ``Collection.indexInformation()`` methods in Typescript.
+
+- Removes support for the ``readConcern`` and ``writeConcern`` options from the
+  ``Collection.listSearchIndexes()`` method. ``listSearchIndexes()`` is an Atlas-specific method, and Atlas
+  search indexes don't support these options.
+
+- Redefines the ``ServerDescription.roundTripTime`` property as a moving average. Previously,
+  it was a weighted average of the most recently observed heartbeat duration and the
+  previous duration.
+
+- You can specify the type of a search index when creating the index, as shown
+  in the following example:
+
+  .. code-block:: js
+     :emphasize-lines: 3
+    
+     const indexName = await collection.createSearchIndex({
+         name: 'my-vector-search-index',
+         type: 'vectorSearch',
+         definition: {
+             mappings: { dynamic: false }
+         }
+     });
+
+- The ``UpdateFilter.$currentDate`` property no longer throws an error when you pass
+  it to a compound method, such as ``findOneAndUpdate()``, on a collection with a limited schema.
+
+- The driver throws a ``MongoTransactionError`` only if you provide a
+  ``ReadPreferenceMode`` other than ``primary`` and then try to perform a command that
+  involves a read operation.
+
+- The data type of the ``TopologyDescription.error`` property is ``MongoError``.
+
+- The ``Collection.indexExists()`` method no longer supports the ``full`` option.
+
+- The ``Collection.indexInformation()``, ``Collection.indexes()``, and
+  ``Db.indexInformation()`` methods have a return type of
+  ``IndexDescriptionCompact | IndexDescriptionInfo[]`` in TypeScript. 
+
+- The driver no longer throws an error when it receives an AWS access key that includes
+  an expiration timestamp.
```

Comment:
somehow fitting in "When fetching KMS credentials" may be worth doing to avoid implying this impacted AWS database auth